woocommerce/woocommerce-ios
# Core Data: Migrate storage usage for TaxStore

Part of #14122 


## Description

This PR updates the usage of storage in TaxStore:
- Replace the deprecated usage of `writerDerivedStorage` with `performAndSave` to ensure all write operations are handled in a single operation queue.
- Unify fetch requests for tax rates in a single fetch for performance optimization.

## Steps to reproduce

1. Testing tax classes
- Log in to a test store and navigate to the Products tab.
- Select an existing product or create a new one.
- Select the Price row > Tax classes.
- Confirm that all tax classes of the store are listed correctly.

2. Testing tax rates
- Ensure that your test store enables tax rates in wp-admin > WooCommerce > Settings > General >  Enable tax rates and calculations.
- Add tax rates in wp-admin > WooCommerce > Settings > Tax > Standard Rates if none exists.
- Log in to the test store in the app.
- Navigate to the Orders tab and select + to add a new order.
- Select one or more product.
- Select Set New Tax Rate.
- Confirm that the correct tax rates are displayed on the list.


## Testing information

Tested with simulator iPhone 16 Pro iOS 18.1 and confirmed that:
- Tax classes are loaded correctly on the Product Price setting screen.
- Tax rates are loaded correctly on the Order creation form.

## Screenshots

Tax class list | Tax rates list
--- | ---
<img src="https://github.com/user-attachments/assets/b2900a1a-70da-4e63-a32d-2ccf99caedb0" width=320 /> | <img src="https://github.com/user-attachments/assets/4ec6a4a0-bc9d-4a61-afd7-8ac7b1d36d5e" width=320 />

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
